File: algo-contest-1-17-07-2022/D/solution.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int queries;
    cin >> queries;
    while (queries--) {
        int n, ans = 0;
        cin >> n;
        vector < int > a(n);
        for (int &item: a)
            cin >> item;

        for (int i = 0; i < n; ++i)
            if (a[i] > i + 1)
                ans = max(ans, a[i] - i - 1);
        cout << ans << "\n";
    }
    return 0;
}

File: algo-contest-1-17-07-2022/D/solution.py
queries = int(input())

while queries:
    queries -= 1
    n = int(input())
    a = list(map(int, input().split()))
    
    ans = 0
    for i in range(0, n):
        if a[i] > i + 1:
            ans = max(ans, a[i] - i - 1)
            
    print(ans)
